sklearn/naive_bayes.py

```diff
@@ -552,11 +552,13 @@ def _check_alpha(self):
                     "alpha should be a scalar or a numpy array with shape [n_features]"
                 )
         if np.min(self.alpha) < _ALPHA_MIN:
-            warnings.warn(
-                "alpha too small will result in numeric errors, setting alpha = %.1e"
-                % _ALPHA_MIN
-            )
-            return np.maximum(self.alpha, _ALPHA_MIN)
+            if not self.force_alpha:
+                warnings.warn(
+                    "alpha too small will result in numeric errors, "
+                    f"setting alpha = {_ALPHA_MIN:.1e}. Use set_alpha ="
+                    "True to keep alpha unchanged."
+                )
+                return np.maximum(self.alpha, _ALPHA_MIN)
```

Comment:
Here is how I envision the deprecation process for `force_alpha`:

Specifically, In 1.1 and 1.2 a warning will always appear when `np.min(self.alpha) < _ALPHA_MIN`. The warning will say that `alpha` will not be changed anymore in 1.3 and the new behavior is enabled by setting `force_alpha=True`.

Also since `force_alpha` is deprecated right away it will also raise a warning saying it will be removed in 1.3.  In other words, `force_alpha="deprecated"` default and we document that the default is `False`. Internally, we will interpret "deprecated" as `False`. For details, see [Maintaining backwards compatibility](https://scikit-learn.org/dev/developers/contributing.html#maintaining-backwards-compatibility) for how to deprecate a parameter.

Reply:
Yeah that makes sense. I would be more than happy to go ahead and add in the depreciation if that's what the reviewers find appropriate. Imo that's the most reasonable solution here.

Reply:
I agree with the steps proposed by @thomasjpfan.

(With the release of 1.1, version numbers need to have their minor number be incremented.)